### PR TITLE
Fixed but with loadCertsFromDir and added tests

### PR DESCRIPTION
**Pull request by diogomonica**
Fixed but with loadCertsFromDir, added tests and more debugging.

@endophage @NathanMcCauley 

Signed-off-by: Diogo Monica <diogo@docker.com>

### CODE REVIEW

File: trustmanager/x509utils.go [hunk 1]
```diff
@@ -118,11 +119,26 @@ func fingerprintCert(cert *x509.Certificate) (CertID, error) {
...
 
 // loadCertsFromDir receives a store AddCertFromFile for each certificate found
-func loadCertsFromDir(s *X509FileStore) {
+func loadCertsFromDir(s *X509FileStore) error {
 	certFiles := s.fileStore.ListFiles(true)
-	for _, c := range certFiles {
-		s.AddCertFromFile(c)
+	for _, f := range certFiles {
+		// ListFiles returns relative paths
+		fullPath := filepath.Join(s.fileStore.BaseDir(), f)
```
Comment (endophage) [inline trustmanager/x509utils.go]: should probably be `s.fileStore.GetPath(f)` given we have the function.

**Comment by endophage** [comment]
LGTM.

Not too concerned either way with my one comment.

File: trustmanager/x509utils.go (hunk 1 above)
Comment (diogomonica) [inline trustmanager/x509utils.go]: Well, GetPath takes a gun so I would have to extract the extension.

So either I do `.Join` with `BaseDir` or I do `GetPath` with a `suffixTrim`.

**Comment by NathanMcCauley** [comment]
LGTM

**Pull request merged**